lua/rest-nvim/utils/init.lua

```diff
@@ -27,11 +27,8 @@ end
 -- file_exists checks if the provided file exists and returns a boolean
 -- @param file File to check
 M.file_exists = function(file)
-  file = io.open(file, "rb")
-  if file then
-    file:close()
-  end
-  return file ~= nil
+  if vim.fn.filereadable(file) ~= 1 then return false end
+  return true
```

Comment:
Maybe we could simplify that with something like this

```suggestion
  return vim.fn.filereadable(file) == 1
```